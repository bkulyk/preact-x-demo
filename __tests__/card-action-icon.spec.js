import React from 'react';
import CardActionIcon from '@bitchin/card-action-icon';
import { shallow } from './enzyme';

it('renders correctly, it is pretty much just a button', () => {
  expect(shallow(<CardActionIcon>View</CardActionIcon>))
    .toMatchSnapshot();
});

it('should send click events to the button', () => {
  expect(shallow(<CardActionIcon onClick={() => {}}>favorite</CardActionIcon>))
    .toMatchSnapshot();
});

it('should pass on any props to the button', () => {
  expect(shallow((
    <CardActionIcon className="something" data-x="x" raised>favorite</CardActionIcon>
  ))).toMatchSnapshot();
});
