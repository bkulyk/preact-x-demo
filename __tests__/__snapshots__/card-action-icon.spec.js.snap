// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly, it is pretty much just a button 1`] = `
<button
  className="mdc-icon-button mdc-card__action mdc-card__action--icon"
  type="button"
>
  <i
    className="material-icons mdc-icon-button__icon"
  >
    View
  </i>
</button>
`;

exports[`should pass on any props to the button 1`] = `
<button
  className="mdc-icon-button mdc-card__action mdc-card__action--icon something"
  data-x="x"
  raised={true}
  type="button"
>
  <i
    className="material-icons mdc-icon-button__icon"
  >
    favorite
  </i>
</button>
`;

exports[`should send click events to the button 1`] = `
<button
  className="mdc-icon-button mdc-card__action mdc-card__action--icon"
  onClick={[Function]}
  type="button"
>
  <i
    className="material-icons mdc-icon-button__icon"
  >
    favorite
  </i>
</button>
`;
